src/scripts/components/DropdownCheckbox.jsx

```diff
@@ -0,0 +1,268 @@
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import { Button, Checkbox, DropdownMenuItem, DropdownMenuItemWild, DropdownMenuScroll, DropdownFilter, DropdownWrapper, Icon } from '../components';
+
+class DropdownCheckbox extends React.Component {
+  state = {
+    activeKey: this.props.activeKey,
+    isOpen: false,
+    hasFilter: false,
+    isCheckbox: this.props.isCheckbox,
+    checked: this.props.isChecked,
+  };
+
+  componentWillMount() {
+    React.Children.map(this.props.children, (child) => {
+      if (!child) return;
+      if (child.type === DropdownFilter) {
+        this.setState({ hasFilter: true });
+      }
+    });
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (newProps.activeKey !== this.props.activeKey) {
+      this.setState({
+        activeKey: newProps.activeKey,
+      });
+    }
+    if (newProps.isChecked !== this.props.isChecked) {
+      this.setState({
+        checked: newProps.isChecked,
+      });
+    }
+  }
+
+  getChildren = () => {
+    let returnChild = null;
+    const { children } = this.props;
+
+    return React.Children.map(children, (child) => {
+      if (!child) return null;
+
+      if (child.type === DropdownMenuItem) {
+        const onClick = () => {
+          if (child.props.id) {
+            if (this.props.onSelect) {
+              this.updateActiveKey(child.props.id);
+              this.props.onSelect(child.props.id);
+            } else {
+              this.updateActiveKey(child.props.id);
+            }
+          }
+
+          if (child.props.onClick) {
+            child.props.onClick();
+          }
+
+          this.handleToggle();
+        };
+
+        returnChild = React.cloneElement(child, {
+          onClick,
+          active: this.state.activeKey && (child.props.id === this.state.activeKey) ? true : false, // eslint-disable-line
+        });
+      } else if (child.type === DropdownFilter) {
+        returnChild = React.cloneElement(child, {
+          onSelect: this.props.onSelect,
+          handleToggle: this.handleToggle,
+          activeKey: this.state.activeKey,
+          icon: this.props.icon,
+          updateActiveKey: this.updateActiveKey,
+        });
+      } else if (child.type === DropdownMenuItemWild) {
+        returnChild = React.cloneElement(child, {
+          handleToggle: this.handleToggle,
+        });
+      } else {
+        returnChild = child;
+      }
+
+      return returnChild;
+    });
+  }
+
+  handleToggle = () => {
+    const $dropdown = ReactDOM.findDOMNode(this.dropdown);
+
+    if (this.state.isOpen) {
```

Comment:
It would be good to destructure here like below:
`const { isOpen } = this.state; `
and then use `isOpen` anywhere you want.